kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/OperatorMain.java

```diff
@@ -66,28 +74,31 @@ void start() {
         operator.installShutdownHook(Duration.ofSeconds(10));
         var registeredController = operator.register(new ProxyReconciler(runtimeDecl()));
         // TODO couple the health of the registeredController to the operator's HTTP healthchecks
+        managementServer.createContext("/metrics", metricsHandler);
+        managementServer.start();
         operator.start();
         LOGGER.info("Operator started.");
     }
 
     void stop() {
         operator.stop();
+        managementServer.stop(0); // TODO maybe this should be configurable
         LOGGER.info("Operator stopped.");
     }
 
     @NonNull
     static RuntimeDecl runtimeDecl() {
         // TODO read these from some configuration CR
-        return new RuntimeDecl(List.of(
-                new FilterApiDecl("filter.kroxylicious.io", "v1alpha1", "KafkaProtocolFilter")));
+        return new RuntimeDecl(List.of(new FilterApiDecl("filter.kroxylicious.io", "v1alpha1", "KafkaProtocolFilter")));
     }
 
     private MicrometerMetrics enablePrometheusMetrics() {
-        registry = new PrometheusMeterRegistry(PrometheusConfig.DEFAULT);
+        final PrometheusMeterRegistry prometheusMeterRegistry = new PrometheusMeterRegistry(PrometheusConfig.DEFAULT);
```

Comment:
This method has a weird combination of updating state and returning a value.

Reply:
Thats a fair point. I'll refactor.

Reply:
Fixed in 969f6f9447b933eeb83c7e728bb289f7c21641ed